File: 189-RotateArray/189-RotateArray.py
# Last updated: 7/31/2025, 5:36:11 PM
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        s=[]
        f=len(nums)-k
        if f>0:
            for i in range(0,f):
                s.append(nums[0])
           
                nums.remove(nums[0])
            nums.extend(s)
            return nums
        else:
            a=abs(len(nums)+f)
            for i in range(0,a):
                s.append(nums[0])
           
                nums.remove(nums[0])
            nums.extend(s)
            return nums

        

File: 189-RotateArray/189-RotateArray.java
// Last updated: 7/31/2025, 5:36:03 PM
class Solution {
    public void rotate(int[] nums, int k) {
        k=k%nums.length;
        int h= nums.length;

        ro(nums, 0, h); 
        ro(nums, 0, k);
        ro(nums, k,h);
    }
    public static void ro(int[] nums, int a,int b){
        int y,j=a,i=b-1;
        while(j<i){
            y=nums[i];
            nums[i]=nums[j];
            nums[j]=y;
            j++;
            i--;
        }
    }
}